File: loops/notes.py
#Zoe Jimenez, Loops notes PYTHON

#What is a loop? 


#What are the two types of loops?
#for loop
nums = [12,3,66,7,3,3,2]

for num in nums:
    print(num)


    #while loops

x = 0
while x < 10:
    print(x)
    x+= 1

#What is iteration
#That specific instance of the loop
#Iteration the amount of times you are looping through


#What are lists? 
num = [1,2,3,4,5,6,7,6]
siblings = ["Alex", "Katie", "Anderw", "Tia"]
print(nums) #prints whole ugly list
print(siblings[3])

siblings[3] = "jake"
print(siblings[3])

siblings.pop(2)
siblings.insert(1, "Jayshree")
siblings.insert(2, ["Joe", "Noah", "Zee"])
print(siblings)

#How do you make lists in python? 
#Use straight braketts, add items with correct data types(strings have quotation makes and interigers have none), they have to have commas between them

#How do you make for loops in python? 
for sibling in siblings:
    print(sibling)

for x in range(1,6,3):
    print(x)

#How do you make while loops in python?
import random
x = 1 #variable to keep count of iteration
goose = random.randint(1,20)

while x <= 20:
    if x == goose:
        print("GOOSE")
        break #stops when goose appears
    else:
        print("Duck..")
    x+= 1

#How do you make lists in C?


#how do you make for loops in C?


#How do you make while loops in C?

File: loops/notes.c
// Zoe Jimenez, Loop notes C

#include <stdio.h>


int main(void){
// While loop
int start = 0;
while(start<5){
    printf("Hello!\n");
    start++;
}
//For loops
int q;
for(q=0;q<5;q++){
    printf("%d\n", q);
}
// What are arrays(lists)
    // an array is a varibale that holds multiple values

// How do you make them?
//array all need to be the same data type!
int grades[] = {88, 97, 100, 70, 72, 99, 89};
// 1. set the data type
// 2. AFTER naming put brackets and write the length of the list
// 3. List is surrounded by curly brackets {}
// 4. Commas, between each items

printf("CSP grade: %d\n", grades[2]);
//change an item in the array
grades[2] = 95;
printf("New CSP grade: %d\n", grades[2]);

//size of list in bytes
int size = sizeof(grades);
// length in list items
int length = sizeof(grades)/sizeof(grades[0]);
printf("The array is %d items long\n", length);

//array with strings
// first brackets sets the length of the array
// second bracket sets length of each string
char movies[][20] = {"Cars", "Treasure Planet", "An American Tale", "Marley and Me", "The Avengers"};
printf("The movie is %s!\n", movies[1]);
int mlength = sizeof(movies)/sizeof(movies[0]);
// How to make a FOR loop in C
// set the iterator, keeps track of times through the loop (best practice to set x or i)
int x;
// in parens (starting point; ending point; count by)
for(x=0;x<=10;x++){
    printf("%d\n", x);
}

int m;
for(m=0;m<mlength;m++){
    printf("%s\n", movies[m]);
}

//How to make a WHILE loop
int w = 100; // start point

while(w>=0){ // stop point
    printf("%d\n", w);
    w -= 10; // what you count by
}


    return 0;
}

